### PR TITLE
TF Lite Exports for YOLO

### PR DESCRIPTION
**Pull request by charleslparker**
@sdesimone / @unmonoqueteclea - I've changed the definition of the tinyYOLOv4 topology to be (hopefully) compatible with TFLite.  At the very least, I'm able to save the model as a tf.lite flatBuffer and reload it in the python TFLite interpreter.  If the iOS one is the same-ish, everything should work out.

See wrappers.tflite_export for the implementation.  I've also put a convenience function in the model wrapper itself.

Two caveats:

1. I had to do some fiddling around to get non-max suppression to work.  It seems like I'm pulling in a "restricted" op, but python's tf lite handles it fine so I'm hoping it's okay.

2. This only works with tensor inputs; not with string paths to image files.  TFLite apparently is unable to read files so that's not going to happen as far as I can tell.

### CODE REVIEW

File: tests/test_pretrained.py [hunk 1]
```diff
@@ -100,11 +108,24 @@ def detect_bounding_boxes(network_name, nboxes, class_list, threshold):
...
+
+    tflite_export(detector, TEST_TF_LITE_MODEL)
+    interpreter = tf.lite.Interpreter(model_path=TEST_TF_LITE_MODEL)
+    interpreter.allocate_tensors()
+
+    assert len(interpreter.get_input_details()) == 1
+    assert len(interpreter.get_output_details()) == 3
+
+    shutil.rmtree(TEST_SAVE_MODEL)
+
```
Comment (unmonoqueteclea) [inline tests/test_pretrained.py:125]: It shouldn't be so difficult adding an image here to perform a prediction. Would it be?

I think some of the errors Sergio was having were at inference time.

**Review by unmonoqueteclea — APPROVED** [review]
Thank you very much, Charlie. 

I am going to train a model now, and @sdesimone will test it when it is ready

File: tests/test_pretrained.py (hunk 1 above)
Comment (unmonoqueteclea) [reply tests/test_pretrained.py:125]: Oh, I see now... Regular TensorFlow op are not supported by Python version of TFLite

Comment (unmonoqueteclea) [reply tests/test_pretrained.py:125]: UPDATE: I tested it with tf-nightly (that supports Tensorflow Ops in Tflite) and it works! Let's see in iOS.

**Comment by charleslparker** [comment]
@unmonoqueteclea / @sdesimone - In the dict of settings you pass when creating the model, you can now pass `output_unfiltered_boxes: True`, which will give you a model that spits out all possible boxes and their associated max-scores / classes.  I'm doing an end-to-end test of inference in tflite in python and everything seems to work.  See the test here:

https://github.com/charleslparker/sensenet/blob/master/tests/test_pretrained.py#L123

Also, if non-max suppression is eliminated, I'm able to convert the model to TFLite without enabling the extended TF operator set, so maybe we can save @sdesimone the hassle of having to recompile TFLite.

Let me know if you get output with this

File: sensenet/models/wrappers.py [hunk 2]
```diff
@@ -48,6 +45,7 @@ def __call__(self, input_data):
 
 class ObjectDetector(object):
     def __init__(self, model, settings):
+        self._unfiltered = settings.output_unfiltered_boxes
```
Comment (unmonoqueteclea) [inline sensenet/models/wrappers.py:48]: When creating the model with 

`wrappers.create_model(meta, settings=SENSENET_CONFIG)`

settings is a dict, so it raises an error:

`AttributeError: 'dict' object has no attribute 'output_unfiltered_boxes'`

I was able to run the conversion changing this line to 

`self._unfiltered = settings["output_unfiltered_boxes"]`

but it won't work when an actual settings object is received...

**Review by unmonoqueteclea — CHANGES_REQUESTED** [review]

File: sensenet/models/wrappers.py (hunk 2 above)
Comment (charleslparker) [reply sensenet/models/wrappers.py:49]: @unmonoqueteclea - I fixed this so you can provide either thing.  Can you check and if it works approve the PR?

**Review by unmonoqueteclea — APPROVED** [review]
Thank you

**Comment by charleslparker** [comment]
@unmonoqueteclea - Apparently I hadn't yet merged this, but this should fix your problem:  You were passing into that function the wrapped model (I think), but it expects you to pass in the actual tf.keras.Model object.  I've created a sort of wrapper in the same name space (`to_tflite`) that will take anything - keras model, wrapper class, raw dict or path to model file.  Please use that (I've also eliminated the instance method in ObjectDetector).  Let me know if it works so I can merge this.

**Comment by unmonoqueteclea** [comment]
Ready to be merged, thank you

**Pull request merged**